examples/systemcalls/systemcalls.c

```diff
@@ -92,6 +146,56 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *
 */
+    // Child_pid
+    int child_pid;
+
+    // Opening outputfile 
+    int fd = open(outputfile, O_WRONLY | O_CREAT | O_TRUNC, 0644); 
+
+    // Check for error on open and return false 
+    if (fd == -1) {
+        perror("open");
+        return false;
+    }
+
+    // Create new child process
+    switch(child_pid = fork()) {
+        case -1: // Error case
+            perror("fork"); 
+            return false;
+
+        case 0: // Child Process
+            if (dup2(fd, 1) < 0) { 
+                perror("dup2"); 
+                exit(1); 
+            }
+            close(fd);
+            execv(command[0], command); 
+            perror("execv"); 
+            exit(1); 
+
+        default:
+            close(fd);
+            int status;
+            pid_t wait_pid = waitpid(child_pid, &status, 0);
+
+            if (wait_pid == -1) {
+                perror("wait");
+                return false;
+            } else {
+                // Check if process exited normally
+                if (WIFEXITED (status)) {
+                    // Checking status of process
+                    if (WEXITSTATUS(status) == 1) {
+                        return false;
+                    } else if (WEXITSTATUS(status) == 0) {
+                        return true;
```

Comment:
@GautamaGandhi think about how you could avoid duplicating logic between the two do_exec functions.

Reply:
@dwalkes thank you for the feedback.